Docs: Update the guide on how to configure SCIM with okta

dmihai opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

This PR includes some minor fixes after following this guide to configure SCIM in okta for a Grafana cloud instance:
- Fix broken links
- Specify the role needed when creating a service account in Grafana
- Add the authentication details when configuring SCIM connection on okta side
- Swap the order of these tabs in the docs: `Integration` and `To App` because when initially setting up SCIM only the `Integration` tab is available. The `To App` tab is visible after the integration is completed.
- Removed the `Troubleshooting` section from the end of the guide because I couldn't find any info about this. We can add it back after the troubleshooting has been added in the docs. 

**Why do we need this feature?**

For improving docs.

**Who is this feature for?**

Users configuring SCIM with okta

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

dmihai committed:
update docs on how to configure SCIM with okta

irenerl24 commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/configure-scim-with-okta/_index.md:
```diff
@@ -19,7 +19,7 @@ weight: 320
 
 {{< admonition type="note" >}}
-Available in [Grafana Enterprise](../../../introduction/grafana-enterprise/) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
+Available in [Grafana Enterprise](/docs/grafana/latest/introduction/grafana-enterprise/) and [Grafana Cloud Advanced](/docs/grafana-cloud/).
```
Available in [Grafana Enterprise](../../../introduction/grafana-enterprise/) and [Grafana Cloud Advanced](/docs/grafana-cloud/).

dmihai replied:
The link `../../../introduction/grafana-enterprise/` didn't work for me on this page https://grafana.com/docs/grafana/next/setup-grafana/configure-security/configure-scim-provisioning/configure-scim-with-okta/.

irenerl24 replied:
try with https://grafana.com/docs/grafana/<GRAFANA_VERSION>/introduction/grafana-enterprise/ instead :)

dmihai committed:
fix enterprise link

irenerl24 committed:
add introductory paragraph under "Configure provisioning settings"

This paragraph has been added to avoid stacked headings that break the flow of the documentation and hurt SEO

irenerl24 reviewed (approved):
LGTM, thank you! :)

dmihai committed:
split SCIM connector base URL config for cloud and onprem

dmihai committed:
Merge branch 'dmihai/scim-okta-docs-update' of github.com:grafana/grafana into dmihai/scim-okta-docs-update

irenerl24 commented:
LGTM, thank you @dmihai!

dmihai commented:
@irenerl24, do we want to backport these changes to v12.0.1? If the answer is yes, I can manually do the backport because the automated process failed.

irenerl24 commented:
no worries, @dmihai, I've backported it already :)
